common/src/main/java/io/druid/metadata/PasswordProvider.java

```diff
@@ -34,5 +34,15 @@
 })
 public interface PasswordProvider
```

Comment:
Right now, it feels like you're wanting to make the PasswordProvider less specific to the metadata store (which I think is nice) but it's still too connected to really work well.